### PR TITLE
Maps block: render Apple Maps maps on WP.com frontend

### PR DESCRIPTION
**Pull request by TimBroddin**
Fixes https://github.com/Automattic/wp-calypso/issues/73645

## Proposed changes:
This PR adds support for rendering Apple Maps on WP.com hosted websites. It does this by setting a `data-source` attribute on the outputted HTML.

Depending on the value of this attribute, a MapBox or Mapkit gets rendered. Currently it's possible to override this behavior and force the rendering of Mapkit, even on a JT site by passing a mapkit query parameter (`?mapkit`). 

CleanShot 2023-02-17 at 13 38 06@2x

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
Discussion: pdDOJh-1tR-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

Apply this diff to your sandbox & sandbox `public-api.wordpress.com`: D101694-code

1. Build the Jetpack plugin
2. Create a post/page with a map
3. View the page, and add `?mapkit` to the URL to force Mapkit to render the map.

### CODE REVIEW

**Comment by mashikag** [comment]
Does this PR have any GH issue that it is resolving?

**Comment by TimBroddin** [comment]
> Does this PR have any GH issue that it is resolving?

Yes, I've edited the description 🙏

File: projects/plugins/jetpack/extensions/blocks/map/map.php [hunk 1]
```diff
@@ -101,7 +100,13 @@ function load_assets( $attr, $content ) {
 
 	Jetpack_Gutenberg::load_assets_as_required( FEATURE_NAME );
 
-	return preg_replace( '/<div /', '<div data-api-key="' . esc_attr( $access_token['key'] ) . '" ', $content, 1 );
+	$source = defined( 'IS_WPCOM' ) && IS_WPCOM ? 'mapkit' : 'mapbox';
+	// phpcs:ignore WordPress.Security.NonceVerification
```
Comment (mashikag) [inline projects/plugins/jetpack/extensions/blocks/map/map.php:104]: Why did we decide to ignore the the `WordPress.Security.NonceVerification` warning? I am sorry if I am being pesky. 🙈

Comment (TimBroddin) [reply projects/plugins/jetpack/extensions/blocks/map/map.php:104]: This is just a debug condition, the production code will lose this, and will read:

```php
if ( $source === 'mapkit' ) {
...
}
```
It seems a bit overkill to create a form with a nonce to enable this debug mode 😀

**Comment by TimBroddin** [comment]
The checks were failing, because Lodash, React & other libraries were being included in the build. This happened because in `utils/index.js` this import happened:

```js
import { isAtomicSite, isSimpleSite } from '../../../../../../js-packages/shared-extension-utils';
```

Both `view/mapbox.js` and `view/mapkit.js` imported `resizeMapContainer` from this file, and even though this doesn't call the `isAtomicSite` or `isSimpleSite` function, no tree shaking is performed on this import. As a result, we ended up with a huge JavaScript bundle.

I fixed this by splitting the functions from `utils` into seperate files. In `index.js` everything is still exported as before, but if you care about bundle size, you can import just the function you want by importing only that file.

```js
import resizeMapContainer from '../utils/resize-map-container';
```

There are still failing PHP tests though, it's unrelated but I'm going to look into them now:

```
  1) WP_Test_Jetpack_Sync_Post::test_delete_post_syncs_event
  Failed asserting that two strings are equal.
  --- Expected
  +++ Actual
  @@ @@
  -'deleted_post'
  +'updated_option'
```

**Comment by TimBroddin** [comment]
continuous-integration/a8c-teamcity is still failing, but it's not related I think. When I look in TeamCity I see that more builds are failing on exactly the same tests.

**Review by mashikag — APPROVED** [review]
The code looks good to me. I have tested the changes on my sandbox environment and all seems to work as expected. 👍

**Comment by TimBroddin** [comment]
Hey @Automattic/jetpack 

I'm working on a feature where we switch from Mapbox to Mapkit (Apple Maps) on WP.com. I'm merging smaller PR's into this [feature branch PR](https://github.com/Automattic/jetpack/pull/29098).

Do you want to review this smaller PR first, before I merge it with the feature branch? I think this will make the final review easier. 

Thank you so much!

**Review by sdixon194 — APPROVED** [review]
Thanks for breaking this down into smaller PRs! I just did a bunch of functional testing on this one and everything worked fine, including existing maps. Also checked self hosted sites too to make sure things continues to work fine.

**Pull request merged**